### DISCUSSION TITLE
inferred type is `str*`; what does the star indicate?

### DISCUSSION THREAD

**Discussion by dkuettel** [microsoft/pyright#4913] [General]
The documentation is very well done, but I could not find any reference to it. The variable in question should be a `str` and it seems it inferred it correctly, but I dont know what `*` means in this context.

**Comment by erictraut** [comment]
The `*` means that the type is [conditioned on a type variable](https://microsoft.github.io/pyright/#/type-concepts-advanced?id=constrained-type-variables-and-conditional-types). It's effectively an intersection of `str` and the type variable from which it was narrowed. That means it can be used in any place where a `str` is valid or the type variable is valid.

**Comment by dkuettel** [comment]
Awesome, thank you. The linked documentation also makes it very clear 🙃 .
